android/src/main/java/com/polidea/flutter_ble_lib/delegate/DeviceConnectionDelegate.java

```diff
@@ -0,0 +1,196 @@
+                        result);
+                return;
+            case MethodName.IS_DEVICE_CONNECTED:
+                isDeviceConnected(deviceId, result);
+                return;
+            case MethodName.OBSERVE_CONNECTION_STATE:
+                observeConnectionState(deviceId, call.<Boolean>argument(ArgumentKey.EMIT_CURRENT_VALUE), result);
+                return;
+            case MethodName.CANCEL_CONNECTION:
+                cancelConnection(deviceId, result);
+                return;
+            default:
+                throw new IllegalStateException("");
+        }
+    }
+
+    private void connectToDevice(String deviceId, Boolean isAutoConnect, Integer requestMtu, Boolean refreshGatt, Long timeoutMillis, @NonNull final MethodChannel.Result result) {
+        RefreshGattMoment refreshGattMoment = null;
+        if (refreshGatt) refreshGattMoment = RefreshGattMoment.ON_CONNECTED;
+
+        final SafeMainThreadResolver safeMainThreadResolver = new SafeMainThreadResolver<>(
+                new OnSuccessCallback<Object>() {
+                    @Override
+                    public void onSuccess(Object data) {
+                        result.success(null);
+                    }
+                },
+                new OnErrorCallback() {
+                    @Override
+                    public void onError(BleError error) {
+                        result.error(String.valueOf(error.errorCode.code), error.reason, null); //TODO @BartoszWilk convert this error
+                    }
+                });
+
+        bleAdapter.connectToDevice(deviceId,
+                new ConnectionOptions(isAutoConnect, requestMtu, refreshGattMoment, timeoutMillis, 0 /* BluetoothGatt.CONNECTION_PRIORITY_BALANCED */),
+                new OnSuccessCallback<Device>() {
+                    @Override
+                    public void onSuccess(Device data) {
+                        safeMainThreadResolver.onSuccess(null);
+                    }
+                }, new OnEventCallback<ConnectionState>() {
+                    @Override
+                    public void onEvent(final ConnectionState data) {
+                        new Handler(Looper.getMainLooper()).post(new Runnable() {
+                            @Override
+                            public void run() {
+                                streamHandler.onNewConnectionState(data);
+                            }
+                        });
+                    }
+                }, new OnErrorCallback() {
+                    @Override
+                    public void onError(final BleError error) {
+                        safeMainThreadResolver.onError(error);
+                    }
+                });
+    }
+
+    private void observeConnectionState(String deviceId, boolean emitCurrentValue, @NonNull final MethodChannel.Result result) {
+        //emit current value if needed; rest is published automatically through connectToDevice()
```

Comment:
1. Do we really need a comment? is there any way to write it down using the code?
2. what about javadoc instead of inner comment

Reply:
I don't how to point in code to the fact that a separate and seemingly not connected method publishes most of the stream content. :)

It doesn't concern the user, so I'm not sure if this should be a javadoc.